dsgrid/config/dataset_schema_handler_base.py

```diff
@@ -126,3 +217,24 @@ def _check_for_duplicates_in_cols(lst: list, table_name: str):
             raise DSGInvalidDataset(f"{table_name} contains duplicated column name(s)={dups}.")
 
         return set(lst)
+
+    def _remap_dimension_columns(self, df, mapping_refs: List[DimensionMappingReferenceModel]):
+        # This will likely become a common activity when running queries.
+        # May need to cache the result. But that will likely be in Project, not here.
+        # This method could be moved elsewhere.
+        for ref in mapping_refs:
+            column = ref.from_dimension_type.value
+            if column in df.columns:
+                mapping_config = self._dimension_mapping_mgr.get_by_id(
+                    ref.mapping_id, version=ref.version
+                )
+                records = models_to_dataframe(mapping_config.model.records)
+                orig = df.count()
+                df = (
+                    df.join(records, df[column] == records.from_id)
```

Comment:
I would add how="left". Default is an inner join (https://spark.apache.org/docs/3.1.1/api/python/reference/api/pyspark.sql.DataFrame.join.html), if `from_id` is partial of a col, df.count() will shrink.

Reply:
In one of our meetings last week we discussed this and decided that we do want an inner join. The mapping should include all records. Note that line 239 guards against this point.

Reply:
Right I saw line 239. If records are already checked to ensure that all from_id exists, then that makes sense.